Add `FRAC_1_SQRT_2PI` constant to f16/f32/f64/f128

sunsided opened the pull request:
This adds the `FRAC_1_SQRT_2PI` to the `f16`, `f32`, `f64` and `f128` as [`1/√(2π)`](https://www.wolframalpha.com/input?i=1%2Fsqrt%282*pi%29). The rationale is that while `FRAC_1_SQRT_PI` already exists, [Gaussian calculations](https://en.wikipedia.org/wiki/Gaussian_function) for random normal distributions require a `1/(σ√(2π))` term, which could then be directly expressed e.g. as `f32::FRAC_1_SQRT_2PI / sigma`.

The actual value is approximately `1/√(2π) = 0.3989422804014326779399460599343818684758586311649346576659258296…`. Truncated/rounded forms were used for the individual types. 

---

~~I did not any of the `#[unstable]` attributes since I am not aware of their implications.~~

**Edit:** I applied the stability attributes from the surrounding types according to what seemed most likely correct. I believe the `more_float_constants` feature marker is incorrectly applied, but I wasn't sure how to proceed.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9141184052/job/25135307656) [(plain)](https://github.com/rust-lang/rust/commit/f5a24b2c1fefb5e55823cdd62eae9c6eb95abf40/checks/25135307656/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 30.2s done
#16 DONE 35.5s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
   Compiling compiler_builtins v0.1.109
error: constant has missing stability attribute
   --> library/core/src/num/f32.rs:331:5
    |
331 |     pub const FRAC_1_SQRT_2PI: f32 = 0.398942280401432677939946059934381868_f32;

error: constant has missing stability attribute
   --> library/core/src/num/f64.rs:331:5
    |
    |
331 |     pub const FRAC_1_SQRT_2PI: f64 = 0.398942280401432677939946059934381868_f64;

error: could not compile `core` (lib) due to 2 previous errors
Build completed unsuccessfully in 0:00:07
  local time: Sat May 18 16:28:44 UTC 2024
```

</details>

sunsided committed:
Add FRAC_1_SQRT_2PI constant to f16/f32/f64/f128

Mark-Simulacrum commented:
New unstable API surface area currently requires [an ACP](https://github.com/rust-lang/libs-team/issues/new?assignees=&labels=api-change-proposal%2C+T-libs-api&projects=&template=api-change-proposal.md&title=%28My+API+Change+Proposal%29). Please file one and if it's approved we can merge.

@rustbot author

sunsided commented:
Thanks, added it at https://github.com/rust-lang/libs-team/issues/383. Exciting!

@rustbot review

tgross35 commented on library/core/src/num/f128.rs:
```diff
@@ -68,6 +68,11 @@ pub mod consts {
         0.564189583547756286948079451560772585844050629328998856844086_f128;
 
+    /// 1/sqrt(2π)
+    #[unstable(feature = "f128", issue = "116909")]
```
For this and f16, please also add a comment to note there is a second feature gate:

```rust
#[unstable(feature = "f128", issue = "116909")]
// #[unstable(feature = "more_float_constants", issue = "103883")]
pub const FRAC_1_SQRT_2PI: /* ... */
```

sunsided committed:
Add FRAC_1_SQRT_2PI doc alias to FRAC_1_SQRT_TAU

This is create symmetry between the already existing TAU constant (2pi)
and the newly-introduced FRAC_1_SQRT_2PI, keeping the more common
name while increasing visibility.

sunsided commented:
I added `#[doc(alias = "FRAC_1_SQRT_TAU")]` to the constants to create a bit of symmetry with the already-existing `TAU` constant, as discussed in rust-lang/libs-team#383.

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 54bb08d53881da7c43b91fd5a6bd8ef9a3b754e6 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 54bb08d53881da7c43b91fd5a6bd8ef9a3b754e6 -->
<!-- homu: {"type":"Approved","sha":"54bb08d53881da7c43b91fd5a6bd8ef9a3b754e6","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
